File: Fibonacci_Pattern/fib.dart
import "dart:io";

int fib(int n) {
  var arr = [];
  int i = 0;
  
  while (i < n + 1){
    if (i == 0 || i == 1){ arr.add(i);}
    else {
      arr.add(arr[i - 1] + arr[i - 2]);
    }
    i += 1;
  }
  return arr[arr.length - 1];
}

int fibo(int n) {
  if (n == 0 || n == 1) {return n;}
  else {
    return fibo(n - 1) + fibo(n - 2);
  }
}

void main() {
  print("Enter the number")
  var data = int.parse(stdin.readLineSync().toString());
  print(fib(data));
  print(fibo(data))
}

File: Fibonacci_Pattern/fib.ts
interface Question {
	value : number
}

function fibonacci(data:Question):number {
	var pattern:Array<number> = new Array()
	var x:number = 0;

	while (x < data.value + 1) {
		if (x == 0 || x == 1) {
			pattern.push(x)
		} else {
			pattern.push(pattern[x - 1] + pattern[x - 2])
		}
		x++;
	}
	return pattern[x - 1]
}
